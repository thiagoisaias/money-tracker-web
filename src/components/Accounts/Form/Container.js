import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import { accountType } from "types";

import Form from "./Form";

import {
  createAccount,
  updateAccount,
  setAccountToEdit,
  clearAccountsError,
  clearAccountToEdit
} from "store/actions/accounts/accounts";

export class Container extends Component {
  static propTypes = {
    accountToEdit: accountType,
    error: PropTypes.string,
    history: PropTypes.object.isRequired,
    isLoading: PropTypes.bool.isRequired,
    match: PropTypes.object.isRequired,
    clearAccountToEdit: PropTypes.func.isRequired,
    createAccount: PropTypes.func.isRequired,
    updateAccount: PropTypes.func.isRequired,
    setAccountToEdit: PropTypes.func.isRequired
  };

  onSubmitData = formData => {
    const {
      accountToEdit,
      history,
      isLoading,
      match,
      createAccount,
      updateAccount
    } = this.props;

    if (isLoading) {
      return;
    }

    if (match.path === "/accounts/new") {
      createAccount(formData, history);
    } else if (match.path === "/accounts/:id/edit" && accountToEdit) {
      updateAccount(formData, accountToEdit.id, history);
    } else {
      return;
    }
  };

  render() {
    const { accountToEdit, error, isLoading, match } = this.props;
    return (
      <Form
        accountToEdit={accountToEdit}
        onSubmitData={this.onSubmitData}
        error={error}
        isLoading={isLoading}
        match={match}
      />
    );
  }

  componentDidMount() {
    const { accountToEdit, history, match } = this.props;
    if (match.path === "/accounts/:id/edit" && accountToEdit === null) {
      history.push("/accounts");
    }
  }

  componentWillUnmount() {
    const { error, match, clearAccountsError, clearAccountToEdit } = this.props;

    if (match.path === "/accounts/:id/edit") {
      clearAccountToEdit();
    }
    if (error) {
      clearAccountsError();
    }
  }
}

const mapStateToProps = state => ({
  accountToEdit: state.accounts.accountToEdit,
  isLoading: state.accounts.isLoading,
  error: state.accounts.error
});

const mapDispatchToProps = {
  createAccount,
  setAccountToEdit,
  clearAccountsError,
  clearAccountToEdit,
  updateAccount
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Container)
);
