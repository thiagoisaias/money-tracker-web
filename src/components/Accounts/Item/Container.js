import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import { accountType } from "types";

import Item from "./Item";

import {
  deleteAccount,
  setAccountToEdit
} from "store/actions/accounts/accounts";

export class Container extends Component {
  static propTypes = {
    accountData: accountType.isRequired,
    handleActiveItem: PropTypes.func.isRequired,
    history: PropTypes.object.isRequired,
    isActive: PropTypes.bool.isRequired,
    isLoading: PropTypes.bool.isRequired,
    deleteAccount: PropTypes.func.isRequired,
    setAccountToEdit: PropTypes.func.isRequired
  };

  handleEdit = () => {
    const { accountData, history, setAccountToEdit } = this.props;
    setAccountToEdit(accountData);
    history.push(`/accounts/${accountData.id}/edit`);
  };

  handleDelete = () => {
    const { accountData, isLoading, deleteAccount } = this.props;
    if (isLoading) {
      return;
    }

    if (!window.confirm("Are you sure you want to delete this account?")) {
      return;
    }
    deleteAccount(accountData.id);
  };

  render() {
    const { accountData, handleActiveItem, isActive } = this.props;
    return (
      <Item
        accountData={accountData}
        handleActiveItem={handleActiveItem}
        handleDelete={this.handleDelete}
        handleEdit={this.handleEdit}
        isActive={isActive}
      />
    );
  }
}

const mapStateToProps = state => ({
  isLoading: state.accounts.isLoading
});

const mapDispatchToProps = {
  deleteAccount,
  setAccountToEdit
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Container)
);
