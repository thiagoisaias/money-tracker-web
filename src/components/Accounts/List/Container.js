import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import { accountType } from "types";

import List from "./List";

import {
  fetchAccounts,
  clearAccountsError
} from "store/actions/accounts/accounts";

export class Container extends Component {
  static propTypes = {
    accountList: PropTypes.arrayOf(accountType).isRequired,
    error: PropTypes.string,
    isLoading: PropTypes.bool.isRequired,
    fetchAccounts: PropTypes.func.isRequired
  };

  componentDidMount() {
    const { fetchAccounts } = this.props;

    fetchAccounts();
  }

  render() {
    const { accountList, error, isLoading } = this.props;
    const propsToPass = {
      accountList,
      error,
      isLoading
    };
    return <List {...propsToPass} />;
  }

  componentWillUnmount() {
    const { error, clearAccountsError } = this.props;

    if (error) {
      clearAccountsError();
    }
  }
}

const mapStateToProps = state => ({
  isLoading: state.accounts.isLoading,
  error: state.accounts.error,
  accountList: state.accounts.accountList
});

const mapDispatchToProps = {
  fetchAccounts,
  clearAccountsError
};

export default connect(mapStateToProps, mapDispatchToProps)(Container);
