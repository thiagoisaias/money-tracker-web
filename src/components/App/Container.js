import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import App from "./App";

export class Container extends Component {
  static propTypes = {
    isAuthenticated: PropTypes.bool.isRequired
  };

  render() {
    return <App isAuthenticated={this.props.isAuthenticated} />;
  }
}

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated
});

export default withRouter(connect(mapStateToProps)(Container));
