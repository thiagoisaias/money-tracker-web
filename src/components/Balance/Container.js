import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import Balance from "./Balance";

import { getOverallBalance } from "store/actions/users/users";

export class Container extends Component {
  render() {
    return <Balance {...this.props} />;
  }

  componentDidMount() {
    this.props.onGetOverallBalance();
  }
}

Container.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  overallBalance: PropTypes.number.isRequired
};

const mapStateToProps = state => ({
  isLoading: state.users.isLoading,
  overallBalance: state.users.overallBalance
});

const mapDispatchToProps = dispatch => ({
  onGetOverallBalance: () => {
    dispatch(getOverallBalance());
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(Container);
