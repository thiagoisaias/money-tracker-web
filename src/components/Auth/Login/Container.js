import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import { clearAuthError, login } from "store/actions/auth/auth";
import Form from "./Form";

export class Container extends Component {
  onSubmitData = formData => {
    const { history, isLoading, onLogin } = this.props;

    if (isLoading) {
      return;
    }

    onLogin(formData, history);
  };

  render() {
    const { onLogin, isLoading, error } = this.props;
    return (
      <Form
        error={error}
        isLoading={isLoading}
        onLogin={onLogin}
        onSubmitData={this.onSubmitData}
      />
    );
  }

  componentWillUnmount() {
    const { error, onClearAuthError } = this.props;
    if (error) {
      onClearAuthError();
    }
  }
}

Container.propTypes = {
  error: PropTypes.string,
  onClearAuthError: PropTypes.func.isRequired,
  onLogin: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  history: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  isLoading: state.auth.isLoading,
  error: state.auth.error
});

const mapDispatchToProps = dispatch => ({
  onClearAuthError: () => {
    dispatch(clearAuthError());
  },

  onLogin: (loginData, history) => {
    dispatch(login(loginData, history));
  }
});

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Container)
);
