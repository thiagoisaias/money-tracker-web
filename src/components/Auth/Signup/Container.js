import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import { signup, clearAuthError } from "store/actions/auth/auth";
import Form from "./Form";

export class Container extends Component {
  onSubmitData = formData => {
    const { history, isLoading, onSignup } = this.props;

    if (isLoading) {
      return;
    }

    onSignup(formData, history);
  };

  render() {
    const { onSignup, isLoading, error } = this.props;
    return (
      <Form
        error={error}
        isLoading={isLoading}
        onSignup={onSignup}
        onSubmitData={this.onSubmitData}
      />
    );
  }

  componentWillUnmount() {
    const { error, onClearAuthError } = this.props;
    if (error) {
      onClearAuthError();
    }
  }
}

Container.propTypes = {
  error: PropTypes.string,
  onClearAuthError: PropTypes.func.isRequired,
  onSignup: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  history: PropTypes.object.isRequired
};

const mapStateToProps = state => {
  return {
    isLoading: state.auth.isLoading,
    error: state.auth.error
  };
};

const mapDispatchToProps = dispatch => {
  return {
    onSignup: (signupData, history) => {
      dispatch(signup(signupData, history));
    },

    onClearAuthError: () => {
      dispatch(clearAuthError());
    }
  };
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Container)
);
