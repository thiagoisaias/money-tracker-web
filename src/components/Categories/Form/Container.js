import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";

import {
  clearCategoryToEdit,
  createCategory,
  updateCategory
} from "store/actions/categories/categories";

import Form from "./Form";

export class Container extends Component {
  onSubmitData = formData => {
    const {
      categoryToEdit,
      history,
      isLoading,
      match,
      onCreateCategory,
      onUpdateCategory
    } = this.props;

    if (isLoading) {
      return;
    }

    if (match.path === "/categories/new") {
      onCreateCategory(formData, history);
    } else if (match.path === "/categories/:id/edit" && categoryToEdit) {
      onUpdateCategory(formData, categoryToEdit.id, history);
    } else {
      return;
    }
  };

  render() {
    const { categoryToEdit, error, isLoading, match } = this.props;
    return (
      <Form
        {...categoryToEdit}
        error={error}
        isLoading={isLoading}
        match={match}
        onSubmitData={this.onSubmitData}
      />
    );
  }

  /* When the page is reloaded and the path is /category/:id/edit, categoryToEdit is null, so the
  app redirects the user back to /categories
  */
  componentDidMount() {
    const { categoryToEdit, history, match } = this.props;
    if (match.path === "/categories/:id/edit" && categoryToEdit === null) {
      history.push("/categories");
    }
  }

  // The value of categoryToEdit should be null when the user leave /categories/:id/edit
  componentWillUnmount() {
    const { match, onClearCategoryToEdit } = this.props;

    if (match.path === "/categories/:id/edit") {
      onClearCategoryToEdit();
    }
  }
}

Container.propTypes = {
  categoryToEdit: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired
  }),
  error: PropTypes.string,
  history: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired,
  match: PropTypes.object.isRequired,
  onClearCategoryToEdit: PropTypes.func.isRequired,
  onCreateCategory: PropTypes.func.isRequired,
  onUpdateCategory: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  categoryToEdit: state.categories.categoryToEdit,
  error: state.categories.error,
  isLoading: state.categories.isLoading
});

const mapDispatchToProps = dispatch => ({
  onClearCategoryToEdit: () => {
    dispatch(clearCategoryToEdit());
  },

  onCreateCategory: (categoryData, history) => {
    dispatch(createCategory(categoryData, history));
  },

  onUpdateCategory: (categoryData, categoryId, history) => {
    dispatch(updateCategory(categoryData, categoryId, history));
  }
});

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(Container)
);
