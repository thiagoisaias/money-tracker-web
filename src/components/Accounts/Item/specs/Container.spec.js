import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Container } from "../Container";

describe("Account Item Container", () => {
  const mockProps = {
    accountData: {
      id: 1,
      name: "Lorem",
      initialBalance: "23423.00",
      currentBalance: "342.34"
    },
    handleActiveItem: jest.fn(),
    history: {},
    isActive: false,
    isLoading: false,
    deleteAccount: jest.fn(),
    setAccountToEdit: jest.fn()
  };
  const component = shallow(<Container {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
