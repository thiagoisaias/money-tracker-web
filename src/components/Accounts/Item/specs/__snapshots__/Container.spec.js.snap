// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account Item Container renders properly 1`] = `
<AccountItem
  accountData={
    Object {
      "currentBalance": "342.34",
      "id": 1,
      "initialBalance": "23423.00",
      "name": "Lorem",
    }
  }
  handleActiveItem={[Function]}
  handleDelete={[Function]}
  handleEdit={[Function]}
  isActive={false}
/>
`;
